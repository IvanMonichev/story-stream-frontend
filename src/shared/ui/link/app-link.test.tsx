import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router'
import { describe, expect, it } from 'vitest'
import AppLink from './app-link'

describe('AppLink', () => {
  const renderWithRouter = (ui: React.ReactNode) =>
    render(<MemoryRouter>{ui}</MemoryRouter>)

  it('renders link with given text', () => {
    renderWithRouter(<AppLink to='/about'>About Us</AppLink>)
    const link = screen.getByRole('link', { name: /about us/i })
    expect(link).toBeInTheDocument()
  })

  it('has correct href', () => {
    renderWithRouter(<AppLink to='/contact'>Contact</AppLink>)
    const link = screen.getByRole('link', { name: /contact/i })
    expect(link).toHaveAttribute('href', '/contact')
  })

  it('applies class from styles', () => {
    renderWithRouter(<AppLink to='/'>Home</AppLink>)
    const link = screen.getByRole('link', { name: /home/i })
    expect(link.className).toMatch(/link/)
  })

  it('matches snapshot', () => {
    const { asFragment } = renderWithRouter(<AppLink to='/'>Home</AppLink>)
    expect(asFragment()).toMatchSnapshot()
  })
})
