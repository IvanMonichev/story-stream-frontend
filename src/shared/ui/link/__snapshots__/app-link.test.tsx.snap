// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppLink > matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="_link_ad8dc7"
    data-discover="true"
    href="/"
  >
    Home
  </a>
</DocumentFragment>
`;
